src/main/java/hudson/remoting/Launcher.java

```diff
@@ -491,19 +636,7 @@ public List<String> parseJnlpArguments() throws ParserConfigurationException, SA
         while (true) {
             URLConnection con = null;
             try {
-                con = Util.openURLConnection(agentJnlpURL);
-                if (con instanceof HttpURLConnection) {
-                    HttpURLConnection http = (HttpURLConnection) con;
-                    if  (agentJnlpCredentials != null) {
-                        String userPassword = agentJnlpCredentials;
-                        String encoding = Base64.getEncoder().encodeToString(userPassword.getBytes(StandardCharsets.UTF_8));
-                        http.setRequestProperty("Authorization", "Basic " + encoding);
-                    }
-                    if (System.getProperty("proxyCredentials", proxyCredentials) != null) {
-                        String encoding = Base64.getEncoder().encodeToString(System.getProperty("proxyCredentials", proxyCredentials).getBytes(StandardCharsets.UTF_8));
```

Comment:
Note loss of interpretation of system property. Fine I think; `Engine` still looks up the system property apparently.

Reply:
Fixed in commit e4c97287.